worlds/ahit/Rules.py

```diff
@@ -0,0 +1,988 @@
+            pos = 0
+        else:
+            pos = world.random.randint(0, lowest_index)
+
+        chapter_list.insert(pos, ChapterIndex.ALPINE)
+
+    if world.is_dlc1() and world.is_dlc2() and final_chapter is not ChapterIndex.METRO:
+        chapter_list.remove(ChapterIndex.METRO)
+        index = chapter_list.index(ChapterIndex.CRUISE)
+        if index >= len(chapter_list):
+            chapter_list.append(ChapterIndex.METRO)
+        else:
+            chapter_list.insert(world.random.randint(index+1, len(chapter_list)), ChapterIndex.METRO)
+
+    lowest_cost: int = world.options.LowestChapterCost.value
+    highest_cost: int = world.options.HighestChapterCost.value
+    cost_increment: int = world.options.ChapterCostIncrement.value
+    min_difference: int = world.options.ChapterCostMinDifference.value
+    last_cost = 0
+    cost = 0
+    loop_count = 0
+
+    for chapter in chapter_list:
+        min_range: int = lowest_cost + (cost_increment * loop_count)
+        if min_range >= highest_cost:
+            min_range = highest_cost-1
+
+        value: int = world.random.randint(min_range, min(highest_cost, max(lowest_cost, last_cost + cost_increment)))
+
+        cost = world.random.randint(value, min(value + cost_increment, highest_cost))
+        if loop_count >= 1:
+            if last_cost + min_difference > cost:
+                cost = last_cost + min_difference
+
+        cost = min(cost, highest_cost)
+        world.set_chapter_cost(chapter, cost)
+        last_cost = cost
+        loop_count += 1
+
+    if final_chapter is not None:
+        world.set_chapter_cost(final_chapter, world.random.randint(
+                                                            world.options.FinalChapterMinCost.value,
+                                                            world.options.FinalChapterMaxCost.value))
+
+    add_rule(world.multiworld.get_entrance("Telescope -> Mafia Town", world.player),
+             lambda state: state.has("Time Piece", world.player, world.get_chapter_cost(ChapterIndex.MAFIA)))
+
+    add_rule(world.multiworld.get_entrance("Telescope -> Battle of the Birds", world.player),
+             lambda state: state.has("Time Piece", world.player, world.get_chapter_cost(ChapterIndex.BIRDS)))
+
+    add_rule(world.multiworld.get_entrance("Telescope -> Subcon Forest", world.player),
+             lambda state: state.has("Time Piece", world.player, world.get_chapter_cost(ChapterIndex.SUBCON)))
+
+    add_rule(world.multiworld.get_entrance("Telescope -> Alpine Skyline", world.player),
+             lambda state: state.has("Time Piece", world.player, world.get_chapter_cost(ChapterIndex.ALPINE)))
+
+    add_rule(world.multiworld.get_entrance("Telescope -> Time's End", world.player),
+             lambda state: state.has("Time Piece", world.player, world.get_chapter_cost(ChapterIndex.FINALE))
+             and can_use_hat(state, world, HatType.BREWING) and can_use_hat(state, world, HatType.DWELLER))
```

Comment:
Not to do a whole logic audit, but, on normal logic (so no beach ball slamming over the fence) doesn't this also require Ice Hat to hit the button on the ceiling?

Reply:
It doesn't, you can just jump from the top of the big dweller block and do a dive to slide under and hit the button. I figured this wasn't too difficult to do or figure out so I decided to have it in the normal logic.

Reply:
Oh, that makes sense. Now that you've said it, I feel like had I ever rolled a seed that required this I likely would've figured it out.